tastypie/resources.py

```diff
@@ -291,7 +293,17 @@ def get_response_class_for_exception(self, request, exception):
         elif isinstance(exception, UnsupportedFormat):
             return http.HttpBadRequest
 
-        return http.HttpApplicationError
+    def handle_patch_for_filefields(self, bundle, data_to_patch):
+        obj = bundle.obj
+        data_to_patch_copy = data_to_patch.copy()
```

Comment:
Entirely unneeded; we aren't modifying `data_to_patch` and we're returning a new dict.
```suggestion
```